hawkular-command-gateway-war/src/main/java/org/hawkular/cmdgw/command/bus/BusQueueProcessors.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright 2015 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.cmdgw.command.bus;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.ListIterator;
+
+import javax.annotation.Resource;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.context.Destroyed;
+import javax.enterprise.context.Initialized;
+import javax.enterprise.event.Observes;
+import javax.inject.Inject;
+import javax.jms.ConnectionFactory;
+import javax.jms.JMSException;
+import javax.websocket.Session;
+
+import org.hawkular.bus.common.BasicMessage;
+import org.hawkular.bus.common.BasicMessageWithExtraData;
+import org.hawkular.bus.common.ConnectionContextFactory;
+import org.hawkular.bus.common.Endpoint;
+import org.hawkular.bus.common.MessageProcessor;
+import org.hawkular.bus.common.consumer.ConsumerConnectionContext;
+import org.hawkular.cmdgw.Constants;
+import org.hawkular.cmdgw.GatewayLoggers;
+import org.hawkular.cmdgw.MsgLogger;
+import org.hawkular.cmdgw.command.ws.ConnectedWsSessions;
+import org.hawkular.cmdgw.command.ws.WsEndpoints;
+import org.hawkular.cmdgw.command.ws.server.FeedWebSocket;
+import org.hawkular.cmdgw.command.ws.server.WebSocketHelper;
+
+/**
+ * A collection of listeners that are attached to bus queues on application start.
+ *
+ * @author <a href="https://github.com/ppalaga">Peter Palaga</a>
+ */
+@ApplicationScoped
+public class BusQueueProcessors {
```

Comment:
call this "BusEndpointProcessors" so we don't lock ourselves into just queues - may want to use topics for some things later on in the future.